### ISSUE TITLE
` Undefined array key "groupName"`  (Elections broken on CiviCRM >=5.47)

### ISSUE THREAD

**Issue by simonwiles** [agileware/au.com.agileware.elections#12]
As of this morning my CiviCRM cron jobs are returning the following line:

```
[PHP Warning] Undefined array key "groupName" at /var/www/html/wp-content/uploads/civicrm/ext/au.com.agileware.elections-master/elections.php:513
```

Any ideas what might be going on here, and how I might debug it?  Thanks!

PHP: 8.0.30
CiviCRM: 5.65.2
Elections: 1.2.3

**Comment by simonwiles** [comment]
The error is caused because the Elections extension is not compatible with the latest CiviCRM code.

https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_alterMailParams/#details
> `$params` array used to include `valueName` and `groupName`. _These should not be referenced_ as they will stop working in future upgrades. The option groups they referred to are obsolete & will also be phased out. `workflow` should be checked instead.

It's not immediately obvious to me how the code should be updated, but this will have to be addressed somehow.

**Comment by agileware-justin** [comment]
What version of PHP is the CLI using?

**Comment by simonwiles** [comment]
Hi!  The CLI's using the same version of PHP as the site: `8.0.30` at time of filing this issue.  I have a test environment running on top of the official WordPress docker container, which is currently build on top of `php:8.0-fpm`.

**Comment by nickperkins** [comment]
@agileware-justin This is related to a change in the hook interface as @simonwiles mentions, not a php versioning issue. It looks like `groupName` should now be `workflow` but I'm not 100% sure.

**Comment by agileware-justin** [comment]
Should be fixed in the 1.2.5 release. Thanks for your patience.
